BAZProject/ViewModels/SearchViewModel.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  SearchViewModel.swift
+//  BAZProject
+//
+//  Created by jmgarciaca on 19/04/23.
+//
+
+import UIKit
+
+/**
+A protocol for making objects searchable with a query string.
+*/
+protocol SearchableProtocol {
+    /*
+    Search for objects matching a given query.
+     - Parameters:
+     - query: A query string used to search for objects.
+     - completion: A closure that will be called once the search is complete.
+    */
+    func search(query: String, completion: @escaping () -> ())
+}
+
+/**
+A view model for searching and displaying a list of movies.
+*/
+class SearchViewModel {
+    private var movies: [Movie] = []
+}
+
+extension SearchViewModel {
+    
+    /**
+     Returns the number of sections in the search results table view.
+     
+     - Returns: The number of sections in the search results table view.
+     */
+    func numberOfSections() -> Int {
+        1
```

Comment:
Puede ser omitido si omites el método de `override func numberOfSections(in tableView: UITableView) -> Int` en la tabla, ya que por default, la tabla siempre tendrá una sección.